### PR TITLE
Extend the sawtooth CLI to support for listing blocks

### PR DESCRIPTION
**Pull request by feihujiang**
#10
This pr implements "sawtooth block list|show [--url URL]"

The command '''#sawtooth block list''' lists the committed block IDs
The command '''#sawtooth block show blockID''' shows the contents of the block 

Signed-off-by: feihujiang <jiangfeihu@huawei.com>

### CODE REVIEW

**Comment by feihujiang** [comment]
@vaporos Could you please review this PR?  This pr implements "sawtooth block list|show [--url URL]" command you suggested in the slack. Looking forward to your feedback, thanks very much. And I'm working on the other commands like "sawtooth store" etc.

File: sawtooth/cli.py [hunk 1]
```diff
@@ -325,6 +418,13 @@ def main(prog_name=os.path.basename(sys.argv[0]), args=sys.argv[1:]):
         do_keygen(args)
     elif args.command == 'submit':
         do_submit(args)
+    elif args.command == 'block':
+        if args.url is not None:
```
Comment (vaporos) [inline sawtooth/cli.py]: Let's put the implementation in a do_block() function similar to do_keygen/do_submit.  We will have a lot of these, making it uniform will be important.

File: sawtooth/cli.py [hunk 2]
```diff
@@ -309,6 +356,52 @@ def do_submit(args):
         if not client.wait_for_commit():
             raise CliException("transaction was not successfully committed")
 
+class SawtoothCtl(_Communication):
```
Comment (vaporos) [inline sawtooth/cli.py]: Would prefer we use SawtoothClient directly, similar to do_submit().  Our intent with the underscore in _Communication is that it be private; at some point we will remove _Communication as we further refine the SawtoothClient class over time, so we are considering it an implementation detail.

**Comment by vaporos** [comment]
We need full names in the Signed-off-by: block.

File: sawtooth/cli.py (hunk 1 above)
Comment (feihujiang) [inline sawtooth/cli.py]: Done

File: sawtooth/cli.py (hunk 2 above)
Comment (feihujiang) [inline sawtooth/cli.py]: Using LedgerWebClient, Done.

**Comment by feihujiang** [comment]
I'm a Chinese person,  FeihuJiang is exactly my full name.

**Comment by feihujiang** [comment]
@vaporos I updated this PR, please take a look.

**Comment by feihujiang** [comment]
And I added "#sawtooth transaction" and "#sawtooth store" commands in this commit.

File: sawtooth/cli.py [hunk 3]
```diff
@@ -25,10 +25,12 @@
...
 from colorlog import ColoredFormatter
 
-from gossip.common import json2dict
-from sawtooth.client import SawtoothClient
+from gossip.common import json2dict, pretty_print_dict
+from sawtooth.client import SawtoothClient, LedgerWebClient
 from sawtooth.exceptions import ClientException
 from sawtooth.exceptions import InvalidTransactionError
+from ledger.transaction.endpoint_registry import EndpointRegistryTransaction
+from __builtin__ import int
```
Comment (trbs) [inline sawtooth/cli.py]: Is this needed ? 

```int``` is already available by default. 
normally __builtin__ is only used when wanting to override the default int object.
In Python3 __builtin__ won't be available as well.

see https://docs.python.org/2.7/reference/executionmodel.html

**Comment by vaporos** [comment]
This needs to pass ./bin/run_lint.  Essentially, run './bin/run_lint' within vagrant, and if it outputs anything, it doesn't pass.  Our overall goal is PEP8 (within reason), and run_lint are the things we enforce from a build system perspective.

`vagrant@vagrant-ubuntu-trusty-64:/project/sawtooth-core$ ./bin/run_lint 
./sawtooth/cli.py:164:1: E302 expected 2 blank lines, found 1
./sawtooth/cli.py:167:80: E501 line too long (87 > 79 characters)
./sawtooth/cli.py:209:1: E302 expected 2 blank lines, found 1
./sawtooth/cli.py:212:80: E501 line too long (87 > 79 characters)
./sawtooth/cli.py:267:1: E302 expected 2 blank lines, found 1
./sawtooth/cli.py:270:80: E501 line too long (87 > 79 characters)
./sawtooth/cli.py:299:1: E302 expected 2 blank lines, found 1
./sawtooth/cli.py:451:1: E302 expected 2 blank lines, found 1
./sawtooth/cli.py:459:1: E302 expected 2 blank lines, found 1
./sawtooth/cli.py:476:24: E272 multiple spaces before keyword
./sawtooth/cli.py:486:1: E302 expected 2 blank lines, found 1
./sawtooth/cli.py:501:80: E501 line too long (81 > 79 characters)
./sawtooth/cli.py:501:82: W291 trailing whitespace
./sawtooth/cli.py:504:24: E272 multiple spaces before keyword
./sawtooth/cli.py:505:80: E501 line too long (91 > 79 characters)
./sawtooth/cli.py:507:80: E501 line too long (81 > 79 characters)
./sawtooth/cli.py:510:80: E501 line too long (86 > 79 characters)
./sawtooth/cli.py:517:1: E302 expected 2 blank lines, found 1
./sawtooth/cli.py:542:80: E501 line too long (86 > 79 characters)
./sawtooth/cli.py:543:80: E501 line too long (92 > 79 characters)
./sawtooth/cli.py:550:1: E302 expected 2 blank lines, found 1
./sawtooth/client.py:573:80: E501 line too long (94 > 79 characters)
./sawtooth/client.py:575:80: E501 line too long (88 > 79 characters)
./sawtooth/client.py:583:80: E501 line too long (82 > 79 characters)
./sawtooth/client.py:715:80: E501 line too long (88 > 79 characters)
./sawtooth/client.py:716:76: W291 trailing whitespace
`

**Comment by vaporos** [comment]
I played around with the various commands here and am really liking this. :)

File: sawtooth/cli.py (hunk 3 above)
Comment (feihujiang) [inline sawtooth/cli.py]: Done, please take a look.

**Comment by feihujiang** [comment]
@vaporos Glad you like this. I  ran  './bin/run_lint', and updated this commit, please review.

**Pull request merged**